File: src/main/java/pl/denisolek/Reservation/ReservationRepository.java
package pl.denisolek.Reservation;

import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

public interface ReservationRepository extends JpaRepository<Reservation, Integer> {
	List<Reservation> findByRestaurantIdAndIsVerified(Integer restaurantId, Boolean isVerified);

	List<Reservation> findByReservationBeginGreaterThanEqualAndReservationEndIsLessThanAndRestaurantIdAndIsVerified(LocalDateTime start, LocalDateTime end, Integer restaurantId, Boolean isVerified);

	List<Reservation> findByDateAndRestaurantIdAndIsVerified(LocalDate date, Integer restaurantId, Boolean isVerified);

	List<Reservation> findByIsVerified(Boolean isVerified);
}

File: src/main/kotlin/pl/denisolek/core/reservation/ReservationRepository.kt
package pl.denisolek.core.reservation

import org.springframework.data.jpa.repository.JpaRepository
import pl.denisolek.core.customer.Customer
import pl.denisolek.core.restaurant.Restaurant

interface ReservationRepository : JpaRepository<Reservation, Int> {
    fun findByRestaurantAndCustomer(restaurantId: Restaurant, customerId: Customer): List<Reservation>
}